taichi/program/sparse_matrix.cpp

```diff
@@ -95,8 +95,9 @@ SparseMatrixBuilder::SparseMatrixBuilder(int rows,
       prog_(prog) {
   auto element_size = data_type_size(dtype);
   TI_ASSERT((element_size == 4 || element_size == 8));
-  ndarray_data_base_ptr_ = std::make_unique<Ndarray>(
-      prog_, dtype_, std::vector<int>{3 * (int)max_num_triplets_ + 1});
+  ndarray_data_base_ptr_ = prog->create_ndarray(
```

Comment:
nit: when your sparsematrixbuilder destructs, you actually want to call `prog->delete_ndarray` to avoid memory leak?